### DISCUSSION TITLE
how do I add all opened buffers when using /buffer?

### DISCUSSION THREAD

**Discussion by PyDataBlog** [olimorris/codecompanion.nvim#514] [Q&A] [answered]
as it states, I want to use all opened buffers as the full context

**Comment by olimorris** [comment, marked as answer]
Manually. If you use telescope, fzf-lua or mini.pick, you can select multiple buffers at once.

**Reply by PyDataBlog** [reply]
> Manually. If you use telescope, fzf-lua or mini.pick, you can select multiple buffers at once.

I use telescope. how would I do that with `/buffer` command?

**Reply by olimorris** [reply]
Whatever your Telescope mapping is to select multiple items in a picker. By default, it's `<tab>`

**Reply by PyDataBlog** [reply]
> Whatever your Telescope mapping is to select multiple items in a picker. By default, it's `<tab>`

Tried tab which had the + markings but when I hit enter after marking all, panel only showed one added item

**Reply by edshamis** [reply]
> > Whatever your Telescope mapping is to select multiple items in a picker. By default, it's `<tab>`
> 
> Tried tab which had the + markings but when I hit enter after marking all, panel only showed one added item

Same experience for me

**Reply by olimorris** [reply]
There will be something in your Telescope.nvim config that's overriding this then.

https://github.com/user-attachments/assets/84fd0e0b-738e-4e27-b648-1e4b0333bd07

My minimal.lua file:

```lua
---@diagnostic disable: missing-fields

--NOTE: Set config path to enable the copilot adapter to work.
--It will search the follwoing paths for the for copilot token:
--  - "$CODECOMPANION_TOKEN_PATH/github-copilot/hosts.json"
--  - "$CODECOMPANION_TOKEN_PATH/github-copilot/apps.json"
vim.env["CODECOMPANION_TOKEN_PATH"] = vim.fn.expand("~/.config")

vim.env.LAZY_STDPATH = ".repro"
load(vim.fn.system("curl -s https://raw.githubusercontent.com/folke/lazy.nvim/main/bootstrap.lua"))()

-- Your CodeCompanion setup
local plugins = {
  {
    "olimorris/codecompanion.nvim",
    dependencies = {
      { "nvim-treesitter/nvim-treesitter", build = ":TSUpdate" },
      { "nvim-lua/plenary.nvim" },
      { "nvim-telescope/telescope.nvim" },
      -- Comment this out if you don't want to setup blink.cmp
      {
        "saghen/blink.cmp",
        lazy = false,
        build = "cargo build --release",
        opts = {
          -- 'enter' for mappings similar to 'super-tab' but with 'enter' to accept
          keymap = {
            preset = "enter",
            ["<S-Tab>"] = { "select_prev", "fallback" },
            ["<Tab>"] = { "select_next", "fallback" },
          },
          sources = {
            completion = {
              enabled_providers = { "lsp", "path", "buffer", "codecompanion" },
            },
            providers = {
              codecompanion = {
                name = "CodeCompanion",
                module = "codecompanion.providers.completion.blink",
                enabled = true,
              },
            },
          },
        },
      },
    },
    opts = {
      --Refer to: https://github.com/olimorris/codecompanion.nvim/blob/main/lua/codecompanion/config.lua
      strategies = {
        --NOTE: Change the adapter as required
        chat = {
          adapter = "copilot",
          slash_commands = {
            ["file"] = {
              opts = {
                provider = "telescope",
              },
            },
          },
        },
        inline = { adapter = "copilot" },
      },
      opts = {
        log_level = "DEBUG",
      },
    },
  },
}

require("lazy.minit").repro({ spec = plugins })

-- Setup Tree-sitter
local ts_status, treesitter = pcall(require, "nvim-treesitter.configs")
if ts_status then
  treesitter.setup({
    ensure_installed = { "lua", "markdown", "markdown_inline", "yaml" },
    highlight = { enable = true },
  })
end
```

**Reply by PyDataBlog** [reply]
Found the solution. Add this for telescope to work:

```lua
					slash_commands = {
						["file"] = {
							opts = {
								provider = "telescope",
							},
						},
						["buffer"] = {
							opts = {
								provider = "telescope",
							},
						},
					},

```

**Reply by olimorris** [reply]
This has been documented in the README and the [config.lua](https://github.com/olimorris/codecompanion.nvim/blob/main/lua/codecompanion/config.lua) file from day one. I strongly recommend reading the former, end to end. I've spent a large chunk of my time trying to make it accessible and detailed for users.

I'd also recommend using the `fetch` slash command to send the [config.lua](https://github.com/olimorris/codecompanion.nvim/blob/main/lua/codecompanion/config.lua) file over to an LLM if you'd like to quickly understand what your options are.

**Comment by PyDataBlog** [comment]
after that the all marked files or buffers were selected. it defaults to dressing ui which fails to select all markings @edshamis @olimorris
